### PR TITLE
added routes.md file

### PR DESCRIPTION
**Pull request by anwilkie**
# Motivation and Context

I have been looking through frontstage and documenting the API endpoints.

# What has changed

Added routes.md file to document API endpoints.

# Links

[Trello Card](https://trello.com/c/ODTURAH3)

### CODE REVIEW

File: frontstage/routes.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+* This endpoint will resend a verification e-mail if the verification token has expired.
+
+`sign-in/logout`
+* This endpoint will sign-out the user from their session.
+
+---
+
+## Surveys endpoints
+
+`surveys/access_survey`
```
[the remark this answers is not in the document]
Comment (rhiGriff) [reply frontstage/routes.md]: Good spot, from what I remember the endpoints should be dashes and not underscores for ease of reading

File: frontstage/routes.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+`surveys/add-survey/add-survey-submit`
+* This endpoint will assign the new survey to the respondent.
+
+`surveys/download_survey`
+* This endpoint will download a survey that the respondent will need to complete.
+
+`surveys/upload_survey`
+* This endpoint will upload a collection instrument to the collection instrument service.
+
+`surveys/upload_failed`
```
[the remark this answers is not in the document]
Comment (anwilkie) [reply frontstage/routes.md]: I have made a card for this issue. It's a little bare-bones at the moment, so you might want to expand on it.

[Trello card](https://trello.com/c/BPu1JJ8x)

File: frontstage/routes.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+To reset a password
+`passwords/reset-password/<token>`
+* POST request to this endpoint allows you to reset your password
+* GET request to this endpoint allows you to get the password reset form
+* `token` is used to be able to access the reset page
+
+`passwords/reset-password/check-email`
+* This endpoint is used to render a check e-mail template.
+* Don't think this endpoint is used.
```
[the remark this answers is not in the document]
Comment (owencjones) [reply frontstage/routes.md]: Can look in the analytics to see if anyone authorised ever hits it?

File: frontstage/routes.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+* POST request to this endpoint allows you to reset your password
+* GET request to this endpoint allows you to get the password reset form
+* `token` is used to be able to access the reset page
+
+`passwords/reset-password/check-email`
+* GET request to this endpoint is used to render a check e-mail template.
+* Don't think this endpoint is used.
+
+`passwords/reset-password/confirmation`
+* GET request to this endpoint renders confirmation page that password's been changed.
```
Comment (owencjones) [inline frontstage/routes.md]: I'd change `password's` to `password has`.  `password's` implies belonging of password, and as a contraction would normally mean `password is` not `password has`.

File: frontstage/routes.md [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+`passwords/resend-password-email-expired-token/<token>`
+* GET request to this endpoint is used to resend an e-mail to a user if their token has expired.
+
+`passwords/forgot-password`
+* GET request to this endpoint will return the forgot password page.
+* POST request to this endpoint will send a reset password request to the party service.
+
+`passwords/forgot-password/check-email`
+* GET request to this endpoint is used to render the check e-mail template for forgot-password.
```
Comment (owencjones) [inline frontstage/routes.md]: Given that `forgot-password` is a route, I'd probably put it in backticks.

**Pull request merged**